### PR TITLE
added example for parse and format input value, see #1297

### PR DESCRIPTION
**Pull request by bodokaiser**
We added a `parseFormatInputValues.tsx` to `examples/` where we show how
one can implement custom parse and format value behavior to, for
example, parse a newline-separated list of emails to an array and format
them back to a newline-separated string.

Questions and suggestions:
* [ ] You probably want to fork [my codesandbox][1] to your account?
* [ ] Is the code style used in `examples/parseFormatInputValue.tsx` correct?
* [ ] Is there away to run `examples/parseFormatInputValue.tsx`?

[1]: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-52dvm

### CODE REVIEW

**Comment by bluebill1049** [comment]
1.  You probably want to fork my codesandbox to your account? 
happy to use yours, just make sure don't get deleted :)

File: examples/parseFormatInputValues.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    onChange(value.split("\n"));
+  };
+
+  return <textarea onChange={handleChange} value={text} />;
+};
+
+export default function App() {
+  const { errors, control, handleSubmit } = useForm();
+  const onSubmit = data => {
+    alert(JSON.stringify(data));
```
Comment (bluebill1049) [inline examples/parseFormatInputValues.tsx:21]: minor: I think `console.log(data)` will do, happy to keep what you got.

File: examples/parseFormatInputValues.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  return (
+    <div className="App">
```
Comment (bluebill1049) [inline examples/parseFormatInputValues.tsx:25]: we can remove this wrapper :) less code for the user to read

File: examples/parseFormatInputValues.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <form onSubmit={handleSubmit(onSubmit)}>
+        {errors.emails && (
```
Comment (bluebill1049) [inline examples/parseFormatInputValues.tsx:27]: We have <ErrorMessage /> component.

**Review by bluebill1049 — CHANGES_REQUESTED** [review]
Awesome stuff! thanks for contributing <3

**Review by bluebill1049 — APPROVED** [review]

**Comment by bodokaiser** [comment]
Thanks for updating and merging the PR! It was a pleasure to contribute :)

**Pull request merged**